contracts/extensions/BasicIssuanceExtension.sol

```diff
@@ -0,0 +1,229 @@
+    }
+
+    /**
+     * ONLY OWNER: Initializes BasicIssuanceExtension to the DelegatedManager.
+     *
+     * @param _delegatedManager     Instance of the DelegatedManager to initialize
+     */
+    function initializeExtension(IDelegatedManager _delegatedManager) external {
+        require(msg.sender == _delegatedManager.owner(), "Must be owner");
+        require(_delegatedManager.isPendingExtension(address(this)), "Extension must be pending");
+
+        ISetToken setToken = _delegatedManager.setToken();
+
+        setManagers[setToken] = _delegatedManager;
+
+        _delegatedManager.initializeExtension();
+
+        ExtensionInitialized(address(setToken), address(_delegatedManager));
+    }
+
+    /**
+     * ONLY OWNER: Initializes BasicIssuanceExtension to the DelegatedManager and BasicIssuanceModule to the SetToken
+     *
+     * @param _delegatedManager             Instance of the DelegatedManager to initialize
+     * @param _maxManagerFee                Maximum fee that can be charged on issue and redeem
+     * @param _managerIssueFee              Fee to charge on issuance
+     * @param _managerRedeemFee             Fee to charge on redemption
+     * @param _feeRecipient                 Address to send fees to
+     * @param _managerIssuanceHook          Instance of the Manager Contract with the Pre-Issuance Hook function
+     */
+    function initializeModuleAndExtension(
+        IDelegatedManager _delegatedManager,
+        uint256 _maxManagerFee,
+        uint256 _managerIssueFee,
+        uint256 _managerRedeemFee,
+        address _feeRecipient,
+        address _managerIssuanceHook
+    ) 
+        external 
+    {
+        require(msg.sender == _delegatedManager.owner(), "Must be owner");
+        require(_delegatedManager.isPendingExtension(address(this)), "Extension must be pending");
+
+        ISetToken setToken = _delegatedManager.setToken();
+
+        require(setToken.isPendingModule(address(issuanceModule)), "BasicIssuanceModule must be pending");
+
+        setManagers[setToken] = _delegatedManager;
+
+        _delegatedManager.initializeExtension();
+
+        bytes memory callData = abi.encodeWithSignature(
+            "initialize(address,uint256,uint256,uint256,address,address)", 
+            setToken,
+            _maxManagerFee,
+            _managerIssueFee,
+            _managerRedeemFee,
+            _feeRecipient,
+            _managerIssuanceHook
+        );
+        _invokeManager(setToken, address(issuanceModule), callData);
+
+        ExtensionInitialized(address(setToken), address(_delegatedManager));
+    }
+
+    /**
+     * ONLY OWNER: Remove an existing SetToken and DelegatedManager tracked by the BasicIssuanceExtension 
+     *
+     * @param _setToken     Instance of the SetToken to remove
+     */
+    function removeExtension(ISetToken _setToken) external override onlyOwner(_setToken) {
```

Comment:
removeExtension should not expose ISetToken as input. it should require from the msg.sender